src/libstd/sys/unix/backtrace/tracing/gcc_s.rs

```diff
@@ -35,28 +35,29 @@ pub fn write(w: &mut Write) -> io::Result<()> {
     unsafe {
         LOCK.lock();
 
-        writeln!(w, "stack backtrace:")?;
-
-        let mut cx = Context { writer: w, last_error: None, idx: 0 };
-        let ret = match {
-            uw::_Unwind_Backtrace(trace_fn,
-                                  &mut cx as *mut Context as *mut libc::c_void)
-        } {
-            uw::_URC_NO_REASON => {
-                match cx.last_error {
-                    Some(err) => Err(err),
-                    None => Ok(())
-                }
+        let mut cx = Context {
+            idx: 0,
+            frames: frames,
+        };
+        let result_unwind = uw::_Unwind_Backtrace(trace_fn,
+                                                  &mut cx as *mut Context
+                                                          as *mut libc::c_void);
+        // See libunwind:src/unwind/Backtrace.c for the return values.
+        // No, there is no doc.
+        let ret = match result_unwind {
+            uw::_URC_END_OF_STACK | uw::_URC_FATAL_PHASE1_ERROR => {
+                Ok((cx.idx, BacktraceContext))
             }
-            _ => Ok(()),
+            _ => unreachable!("\"{:?}\" should not be triggered",
+                              result_unwind),
```

Comment:
Er sorry, what I mean is that this *must* not panic. It should return an error of some kind (`Err(())` is fine) so the caller can deal with the failure.